ironfish-rust/src/public_account/description.rs

```diff
@@ -0,0 +1,275 @@
+        for _ in 0..signers_len {
+            let mut signer_buf = [0; 32];
+            reader.read_exact(&mut signer_buf)?;
+            let signer = VerifyingKey::from_bytes(&signer_buf)
+                .map_err(|_| IronfishError::new(IronfishErrorKind::InvalidData))?;
+            signers.push(signer);
+        }
+
+        let mut signatures_len_buf = [0; 2];
+        reader.read_exact(&mut signatures_len_buf)?;
+        let signatures_len = u16::from_le_bytes(signatures_len_buf) as usize;
+
+        let mut signatures = Vec::with_capacity(signatures_len);
+        for _ in 0..signatures_len {
+            let mut signature_buf = [0; 64];
+            reader.read_exact(&mut signature_buf)?;
+            let signature = Signature::from_bytes(&signature_buf);
+            signatures.push(signature);
+        }
+
+        let mut transfers_len_buf = [0; 2];
+        reader.read_exact(&mut transfers_len_buf)?;
+        let transfers_len = u16::from_le_bytes(transfers_len_buf) as usize;
+
+        let mut transfers = Vec::with_capacity(transfers_len);
+        for _ in 0..transfers_len {
+            let transfer = Transfer::read(&mut reader)?;
+            transfers.push(transfer);
+        }
+        let address = PublicAddress::read(&mut reader)?;
+
+        let description = Self {
+            version,
+            min_signers,
+            signers,
+            signatures,
+            address,
+            transfers,
+        };
+        description.valid()?;
+        Ok(description)
+    }
+
+    pub fn write<W: io::Write>(&self, mut writer: W) -> Result<(), IronfishError> {
+        // TODO: think about ideal ordering here
+        writer.write_all(&self.version.to_le_bytes())?;
+
+        writer.write_all(&self.min_signers.to_le_bytes())?;
+
+        let signers_len: u16 = self.signers.len().try_into()?;
+        writer.write_all(&signers_len.to_le_bytes())?;
+
+        for signer in &self.signers {
+            writer.write_all(signer.as_bytes())?;
+        }
+
+        let signatures_len: u16 = self.signatures.len().try_into()?;
+        writer.write_all(&signatures_len.to_le_bytes())?;
+
+        for signature in &self.signatures {
+            writer.write_all(&signature.to_bytes())?;
+        }
+
+        let transfers_len: u16 = self.transfers.len().try_into()?;
+        writer.write_all(&transfers_len.to_le_bytes())?;
+
+        for transfer in &self.transfers {
+            transfer.write(&mut writer)?;
+        }
+
+        self.address.write(&mut writer)?;
+
+        Ok(())
+    }
+
+    fn valid(&self) -> Result<(), IronfishError> {
+        if self.min_signers < 1 {
+            return Err(IronfishError::new(IronfishErrorKind::InvalidThreshold));
+        }
+
+        if self.signers.len() < self.min_signers as usize {
+            return Err(IronfishError::new(IronfishErrorKind::InvalidThreshold));
+        }
+
+        let unique_signers: Vec<_> = self.signers.iter().collect();
```

Comment:
This is not removing any duplicates, it's only creating a copy of `self.signers` in a slower way than `.clone()`.

I think you meant something like:

```rust
let mut unique_signers = self.signers.clone();
unique_signers.sort_unstable();
unique_signers.dedup();
```

Reply:
I meant to write:
```
let unique_signers: HashSet<_> = self.signers.iter().collect();
```
But your solution below involving the signer check is clearly better, I will go with that.